examples/ffmpeg_video_encoding_ex.cpp

```diff
@@ -0,0 +1,127 @@
+// The contents of this file are in the public domain. See LICENSE_FOR_EXAMPLE_PROGRAMS.txt
+/*
+
+    This is an example illustrating the use of the ffmpeg wrappers, in this case the encding API.
+
+    This is a pretty simple example. It loads a video file, extracts the images and
+    re-encodes them into a raw buffer using a user-specified codec.
+    
+    Please see the following examples on how to decode, demux, and get information on your installation of ffmpeg:
+        - ffmpeg_info_ex.cpp
+        - ffmpeg_video_decoding_ex.cpp
+        - ffmpeg_video_demuxing_ex.cpp
+*/
+
+#include <cstdio>
+#include <dlib/media.h>
+#include <dlib/cmd_line_parser.h>
+
+using namespace std;
+using namespace dlib;
+using namespace dlib::ffmpeg;
+
+int main(const int argc, const char** argv)
+try
+{
+    command_line_parser parser;
+    parser.add_option("i",      "input video", 1);
+    parser.add_option("codec",  "codec name. e.g. h264. Defaults to mpeg4", 1);
+    parser.add_option("height", "height of encoded stream. Defaults to whatever is in the video file", 1);
+    parser.add_option("width",  "width of encoded stream. Defaults to whatever is in the video file", 1);
+
+    parser.set_group_name("Help Options");
+    parser.add_option("h",      "alias of --help");
+    parser.add_option("help",   "display this message and exit");
+
+    parser.parse(argc, argv);
+    const char* one_time_opts[] = {"i", "codec", "height", "width"};
+    parser.check_one_time_options(one_time_opts);
+
+    if (parser.option("h") || parser.option("help"))
+    {
+        parser.print_options();
+        return 0;
+    }
+
+    if (!parser.option("i"))
+    {
+        cout << "Missing -i" << endl;
+        parser.print_options();
+        return 0;
+    }
+
+    const std::string filepath = get_option(parser, "i", "");
```

Comment:
Here I would've just used this, since we already made sure there's an argument. But maybe that's just nitpicking :P.
```suggestion
    const std::string filepath = parser.option("i").argument();
```

Reply:
I don't mind. I don't use dlib's command line option stuff much so I'll do whatever is best.